hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -266,8 +267,27 @@ public void run() {
       throw new SchedulerException(msg, e);
     }
 
-    MesosSchedulerDriver driver = new MesosSchedulerDriver(this,
-      frameworkInfo.build(), hdfsFrameworkConfig.getMesosMasterUri());
+    MesosSchedulerDriver driver;
+
```

Comment:
Let's break this into a separate method. This scheduler class definitely needs to be broken down and requires some refactoring, but let's try to break it down as we go.